nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExtractGrok.java

```diff
@@ -70,33 +65,51 @@
 @Tags({"grok", "log", "text", "parse", "delimit", "extract"})
 @CapabilityDescription("Evaluates one or more Grok Expressions against the content of a FlowFile, " +
     "adding the results as attributes or replacing the content of the FlowFile with a JSON " +
-    "notation of the matched content")
+    "notation of the matched content\n" +
+        "uses https://github.com/thekrakken/java-grok.")
 @WritesAttributes({
-    @WritesAttribute(attribute = "grok.XXX", description = "When operating in flowfile-attribute mode, each of the Grok identifier that is matched in the flowfile " +
-        "will be added as an attribute, prefixed with \"grok.\" For example," +
-        "if the grok identifier \"timestamp\" is matched, then the value will be added to an attribute named \"grok.timestamp\"")})
+    @WritesAttribute(attribute = "{result prefix}XXX", description = "When operating in flowfile-attribute mode, each of the Grok identifier that is matched in the flowfile " +
+        "will be added as an attribute, prefixed with \"{result prefix}\" For example," +
+        "if the grok identifier \"timestamp\" is matched, then the value will be added to an attribute named \"{result prefix}timestamp\""),
+
+        @WritesAttribute(attribute = "ExtractGrok.exception", description = "if an error occurs, an exception will be written to this attribute, " +
+                "and the flow routed to 'unmatched' ")
```

Comment:
This should read "and the FlowFile routed" rather than "and the flow routed". In such a case, though, we should not be routing to 'unmatched' but rather we should route to a failure relationship.